src/Tizen.TV.UIControls.Forms/Renderer/AnimatedNavigationPageRenderer.cs

```diff
@@ -0,0 +1,226 @@
+                }
+            }
+            base.Dispose(disposing);
+        }
+
+        protected override void OnElementChanged(ElementChangedEventArgs<AnimatedNavigationPage> e)
+        {
+            if (_navigationStack == null)
+            {
+                _navigationStack = new NavigationStack(XForms.NativeParent);
+                _navigationStack.Show();
+                _navigationStack.SetAlignment(-1, -1);
+                _navigationStack.SetWeight(1, 1);
+                _navigationStack.Show();
+                SetNativeView(_navigationStack);
+            }
+
+            if (e.OldElement != null)
+            {
+                var navigation = e.OldElement as INavigationPageController;
+                navigation.PopRequested -= OnPopRequested;
+                navigation.PopToRootRequested -= OnPopToRootRequested;
+                navigation.PushRequested -= OnPushRequested;
+                navigation.RemovePageRequested -= OnRemovePageRequested;
+                navigation.InsertPageBeforeRequested -= OnInsertPageBeforeRequested;
+            }
+
+            if (e.NewElement != null)
+            {
+                var navigation = e.NewElement as INavigationPageController;
+                navigation.PopRequested += OnPopRequested;
+                navigation.PopToRootRequested += OnPopToRootRequested;
+                navigation.PushRequested += OnPushRequested;
+                navigation.RemovePageRequested += OnRemovePageRequested;
+                navigation.InsertPageBeforeRequested += OnInsertPageBeforeRequested;
+                _previousPage = e.NewElement.CurrentPage;
+            }
+
+            base.OnElementChanged(e);
+        }
+
+        protected override void OnElementReady()
+        {
+            base.OnElementReady();
+            var pageController = Element as IPageController;
+
+            foreach (Page page in pageController.InternalChildren)
+            {
+                _navigationStack.Push(Platform.GetOrCreateRenderer(page).NativeView);
+            }
+        }
+
+        protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            base.OnElementPropertyChanged(sender, e);
+
+            if (e.PropertyName == NavigationPage.CurrentPageProperty.PropertyName)
+            {
+                Device.BeginInvokeOnMainThread(() =>
+                {
+                    (_previousPage as IPageController)?.SendDisappearing();
+                    _previousPage = Element.CurrentPage;
+                    (_previousPage as IPageController)?.SendAppearing();
+                });
+            }
+        }
+
+        void OnPushRequested(object sender, NavigationRequestedEventArgs nre)
+        {
+            var renderer = Platform.GetOrCreateRenderer(nre.Page);
+            var animation = CurrentPage.GetPushAnimation();
+            var animated = (nre.Animated || _navigationStack.Stack.Count == 0) && animation != null;
```

Comment:
The first item on Stack can't animate, 

```suggestion
            var animated = nre.Animated && _navigationStack.Stack.Count != 0 && animation != null;
```

Reply:
The root page included when the `AnimatedNavigationPage` is created is already set so that it is not animated. And, the first item (page) added later can be animated. However, the conditions have been modified to be simpler. 😉